In [4]:
import pandas as pd
import numpy as np
from time import time
from datetime import datetime
import pymysql
import pandas.io.sql as psql
import sqlalchemy
from datetime import date
import json
import math
from datetime import timedelta
import matplotlib.pyplot as plt
import matplotlib
import numpy as np
from matplotlib import colors
from matplotlib.ticker import PercentFormatter

# Fixing random state for reproducibility
np.random.seed(19680801)
matplotlib.rcParams['figure.figsize'] = [30, 30]

print(__doc__)

Automatically created module for IPython interactive environment


In [5]:
df = pd.read_csv('rewards_redeemed_data.csv')

In [12]:
len(df)

19396

### Average user has redeemed awards only once

In [9]:
df.reward_redeemed.mean()

1.4780882656217778

### Max number of rewards

In [10]:
df.reward_redeemed.max()

27

In [13]:
len(df[df.reward_redeemed >= 7])

195

### Out of almost 20,000 redemptions, only 195 customers have redeemed 7 or more times

In [14]:
df.rewards_redeemed_value.mean()

1780.7900598061456